official/resnet/resnet_run_loop.py

```diff
@@ -384,16 +405,37 @@ def input_fn_eval():
       benchmark_logger = logger.BenchmarkLogger(flags.benchmark_log_dir)
       benchmark_logger.log_estimator_evaluation_result(eval_results)
 
+  # Return the classifier after all epochs are completed.
+  return classifier
+
+
+def export_savedmodel(classifier, export_dir, shape, batch_size):
+  """Exports a saved model for the given classifier."""
+  input_receiver_fn = export.build_tensor_serving_input_receiver_fn(
+      shape, batch_size=batch_size)
+  classifier.export_savedmodel(export_dir, input_receiver_fn)
+  return classifier
+
+
+def warn_on_multi_gpu_export(multi_gpu=False):
+  """For the time being, multi-GPU mode does not play nicely with exporting."""
+  if multi_gpu:
+    tf.logging.warning(
+        'You are exporting a SavedModel while in multi-GPU mode. Note that '
+        'the resulting SavedModel will require the same GPUs be available.'
+        'If you wish to serve the SavedModel from a different device, '
+        'try exporting the SavedModel with multi-GPU mode turned off.')
+
```

Comment:
FYI @isaprykin - For now, warning the user. Eventually, it would be nice if Estimator would know not to run the saved model through the replication part of the graph.

Reply:
Checkpoint loading seems flexible enough that you can load between single and multi gpu models. So in the case of multi gpu models it might makes sense to construct a new estimator, load in the trained weights, and then serialize that. It's perhaps not the most elegant solution, but training with multi_gpu and then serving on a per-GPU basis seems like the most common use case so supporting it is somewhat important.

Reply:
For now, I'm going to leave as is. DistributionStrategies is a moving target that aims to hide replicate_model_fn, so we can reevaluate in a ~month when that is firmed up.

Reply:
That sounds perfectly reasonable.